ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala

```diff
@@ -3,35 +3,44 @@
 
 package com.daml.ledger.participant.state.v1
 
-import java.time.Instant
-
 /** Information provided by the submitter of changes submitted to the ledger.
   *
   * Note that this is used for party-originating changes only. They are
   * usually issued via the Ledger API.
   *
-  * @param actAs: the non-empty set of parties that submitted the change.
+  * @param actAs the non-empty set of parties that submitted the change.
   *
-  * @param applicationId: an identifier for the DAML application that
+  * @param applicationId an identifier for the DAML application that
   *   submitted the command. This is used for monitoring and to allow DAML
   *   applications subscribe to their own submissions only.
   *
-  * @param commandId: a submitter provided identifier that he can use to
-  *   correlate the stream of changes to the participant state with the
-  *   changes he submitted.
-  *
-  * @param deduplicateUntil: the time until which the command should be deduplicated.
-  *   Command deduplication is already performed by the participant.
-  *   The ledger may choose to perform additional (cross-participant)
-  *   command deduplication. If it chooses to do so, it must follow the
-  *   same rules as the participant:
-  *   - Deduplication is based on the (submitter, commandId) tuple.
-  *   - Commands must not be deduplicated after the deduplicateUntil time has passed.
-  *   - Commands should not be deduplicated after the command was rejected.
+  * @param commandId a submitter-provided identifier to identify an intended ledger change
+  *   within all the submissions by the same parties and application.
+  *
+  * @param deduplicationPeriod The deduplication period for the command submission.
+  *   Used for the deduplication guarantee described in the [[ReadService.stateUpdates]].
+  *
+  * @param submissionId An identifier for the submission that allows an application
+  *   to correlate completions to its submissions.
+  *
+  * @param submissionRank The rank of the submission among all submissions with the same change ID.
+  *   Used for the submission rank guarantee described in the [[ReadService.stateUpdates]].
   */
 final case class SubmitterInfo(
     actAs: List[Party],
     applicationId: ApplicationId,
     commandId: CommandId,
-    deduplicateUntil: Instant,
-)
+    deduplicationPeriod: DeduplicationPeriod,
+    submissionId: SubmissionId,
+    submissionRank: Offset,
+) {
+
+  /** The ID for the ledger change */
+  def changeId: ChangeId = ChangeId(applicationId, commandId, actAs = actAs.toSet)
+}
+
+/** Identifier for ledger changes used by command deduplication
+  *
+  * @see ReadService.stateUpdates for the command deduplication guarantee
+  */
+case class ChangeId(applicationId: ApplicationId, commandId: CommandId, actAs: Set[Party])
```

Comment:
```suggestion
// TODO v2: change to a `SortedSet` if possible to ensure a canonical representation of the ChangeId
case class ChangeId(applicationId: ApplicationId, commandId: CommandId, actAs: Set[Party])
```

Reply:
`SortedSet` looks like an implementation concern of kvutils. We hardly use `SortedSet` in Canton, so I'm hesitant to enforce it here. All I care about is the equality of `ChangeId`s, which we need for the specification of the `ReadService`. How about making the fields `private`?

Reply:
Making them private makes this class almost unusable. Alternatively, we could add a comment for implementations that they should ensure canonical serialization.

Reply:
TODO: add comments
Should offer stable hash for actAs that we can store in the DB. Maybe even use the hash for equality?

Reply:
Added the interface.